plat/drivers/virtio/virtio_pci.c

```diff
@@ -212,35 +212,24 @@ static void vpci_legacy_vq_release(struct virtio_dev *vdev,
 	virtqueue_destroy(vq, a);
 }
 
-static int vpci_legacy_pci_vq_find(struct virtio_dev *vdev, __u16 num_vqs,
+/* TODO: change documentation if available */
+static int vpci_legacy_pci_vq_find(struct virtio_dev *vdev, __u16 vq_id,
 				   __u16 *qdesc_size)
 {
 	struct virtio_pci_dev *vpdev = NULL;
-	int vq_cnt = 0, i = 0, rc = 0;
 
 	UK_ASSERT(vdev);
 	vpdev = to_virtiopcidev(vdev);
 
-	/* Registering the interrupt for the queue */
-	rc = ukplat_irq_register(vpdev->pdev->irq, virtio_pci_handle, vpdev);
-	if (rc != 0) {
-		uk_pr_err("Failed to register the interrupt\n");
-		return rc;
-	}
+	virtio_cwrite16((void *) (unsigned long)vpdev->pci_base_addr,
+			VIRTIO_PCI_QUEUE_SEL, vq_id);
+	*qdesc_size = virtio_cread16(
+			(void *) (unsigned long)vpdev->pci_base_addr,
+			VIRTIO_PCI_QUEUE_SIZE);
+	if (unlikely(!*qdesc_size))
+		uk_pr_err("Virtqueue %d not available\n", vq_id);
```

Comment:
Thanks for finding the bug. Traditionally it returned the no of queues found. Hence, instead of the -ENOENT function should return 0 when it fails to find the queue.